mmdet3d/core/bbox/coders/monoflex_bbox_coder.py

```diff
@@ -0,0 +1,366 @@
+
+        # predict the uncertainties of the solved depths from
+        # groups of keypoints
+
+        pred_corner_offset_uncertainty = \
+            reg[:, self.key2channel('corner_uncertainty')]
+
+        if self.use_combined_depth:
+            pred_combined_uncertainty = torch.cat(
+                (pred_depth_uncertainty.unsqueeze(-1),
+                 pred_corner_offset_uncertainty),
+                dim=1).exp()
+            pred_combined_depth = torch.cat(
+                (pred_direct_depth.unsqueeze(-1), pred_keypoints_depth), dim=1)
+            pred_uncertainty_weights = 1 / pred_combined_uncertainty
+            pred_uncertainty_weights = \
+                pred_uncertainty_weights / \
+                pred_uncertainty_weights.sum(dim=1, keepdim=True)
+            pred_corner_depth = torch.sum(
+                pred_combined_depth * pred_uncertainty_weights, dim=1)
+
+        # compute the corners
+        pred_locations = self.anno_encoder.decode_location_flatten(
+            points, pred_offsets2d, pred_corner_depth, cam2imgs, batch_inds)
+        # decode yaws and alphas
+        pred_yaws, _ = self.anno_encoder.decode_axes_orientation(
+            pred_orientations, pred_locations)
+        # decode corners
+        pred_corners = self.anno_encoder.encode_box3d(pred_yaws,
+                                                      pred_dimensions,
+                                                      pred_locations)
+
+        return (pred_locations, pred_dimensions, pred_orientations,
+                pred_corners)
+
+    def decode_depth(self, depths_offset):
+        """Transform depth offset to depth."""
+        if self.depth_mode == 'exp':
+            depth = depths_offset.exp()
+        elif self.depth_mode == 'linear':
+            depth = depths_offset * self.depth_ref[1] + self.depth_ref[0]
+        elif self.depth_mode == 'inv_sigmoid':
+            depth = 1 / torch.sigmoid(depths_offset) - 1
+        else:
+            raise ValueError
+
+        if self.depth_range is not None:
+            depth = torch.clamp(
+                depth, min=self.depth_range[0], max=self.depth_range[1])
+
+        return depth
+
+    def decode_location_flatten(self, points, offsets, depths, calibs,
+                                pad_size, batch_idxs):
+        gts = torch.unique(batch_idxs, sorted=True).tolist()
+        locations = points.new_zeros(points.shape[0], 3).float()
+        points = (points + offsets) * self.down_ratio - pad_size[batch_idxs]
+
+        for idx, gt in enumerate(gts):
+            corr_pts_idx = torch.nonzero(batch_idxs == gt).squeeze(-1)
+            calib = calibs[gt]
+            # concatenate uv with depth
+            corr_pts_depth = torch.cat(
+                (points[corr_pts_idx], depths[corr_pts_idx, None]), dim=1)
+            locations[corr_pts_idx] = calib.project_image_to_rect(
+                corr_pts_depth)
+
+        return locations
+
+    def decode_depth_from_keypoints(self,
```

Comment:
I found this function has not been utilized. Whether this function is utilized somewhere else. If not, this function may need to be discarded.